pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyClientCnx.java

```diff
@@ -54,10 +66,33 @@ protected ByteBuf newConnectCommand() throws Exception {
 
         authenticationDataProvider = authentication.getAuthData(remoteHostName);
         AuthData authData = authenticationDataProvider.authenticate(AuthData.INIT_AUTH_DATA);
-        return Commands.newConnect(authentication.getAuthMethodName(), authData, this.protocolVersion,
-            PulsarVersion.getVersion(), proxyToTargetBrokerAddress, clientAuthRole, clientAuthData,
-            clientAuthMethod);
+        return Commands.newConnect(authentication.getAuthMethodName(), authData, protocolVersion,
+                PulsarVersion.getVersion(), proxyToTargetBrokerAddress, clientAuthRole, clientAuthData,
+                clientAuthMethod);
     }
 
-    private static final Logger log = LoggerFactory.getLogger(ProxyClientCnx.class);
+    @Override
+    protected void handleAuthChallenge(CommandAuthChallenge authChallenge) {
+        checkArgument(authChallenge.hasChallenge());
+        checkArgument(authChallenge.getChallenge().hasAuthData());
+
+        boolean isRefresh = Arrays.equals(AuthData.REFRESH_AUTH_DATA_BYTES, authChallenge.getChallenge().getAuthData());
+        if (!forwardClientAuthData || !isRefresh || refreshClientAuthDataNotifier == null) {
+            super.handleAuthChallenge(authChallenge);
+            return;
+        }
+
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("{} Request to refresh the original client authentication data", ctx.channel());
+            }
+            refreshClientAuthDataNotifier.run();
```

Comment:
Looks like we can only pass the `ctx` and `protocolVersion` to `ProxyClientCnx` from `ProxyConnection`, so we don't need a callback mechanism to complete the auth challenge handling

Reply:
Yes, when authentication data is refreshed, we send an `authResponse` command to the broker that the proxy client connected.

Reply:
@nodece I mean we can remove the `refreshClientAuthDataNotifier`, instead add `ctx` and `protocolVersion` to `ProxyClientCnx`

Reply:
Done.